src/engine/Join.cpp

```diff
@@ -144,6 +144,21 @@ void Join::computeResult(ResultTable* result) {
                     _leftJoinCol, rightRes->_idTable, _rightJoinCol,
                     &result->_idTable);
 
+  // If only one of the two operands has a local vocab, pass it on.
+  bool leftLocalVocabEmpty = leftRes->_localVocab->empty();
+  bool rightLocalVocabEmpty = rightRes->_localVocab->empty();
+  if (!leftLocalVocabEmpty || !rightLocalVocabEmpty) {
```

Comment:
This can be made a function in the Operation class, s.t. this becomes:

`result->_localVocab = getLocalVocabIfThereIsAtMostOne({leftRes, rightRes}, "possibly customized error message")`
In the case that none of them is set, you can just create a new, empty one (the overhead of the `shared_ptr` is not an issue here).

Reply:
Once you have such a function, it should be easy to also add this to other operations.

Reply:
Done, it's now a static function in the LocalVocab class. It's anyway just a temporary solution.